tests/test_event_pub.py

```diff
@@ -54,27 +60,49 @@ async def _publish_validated(
 event_recorder = EventRecorder()
 
 
-notification_emitter = NotificationEmitter(
+event_publisher = EventPubTranslator(
     config=dummy_config, event_publisher=event_recorder
 )
 
 
-async def test_sending_a_notification():
-    """Test that a notification is translated properly."""
-    await notification_emitter.notify(
-        email="someone@somewhere.org",
-        full_name="Some User Name",
-        subject="Some subject",
-        text="Some text",
+@pytest.mark.parametrize("status", ["created", "allowed", "denied"])
+async def test_publishing_an_event(status: str):
+    """Test that an event is published properly."""
+    request = AccessRequest(
+        id="unique_access_request_id",
+        user_id="user123",
+        dataset_id="dataset456",
+        email="requester@example.com",
+        request_text="Requesting access for research purposes",
+        access_starts=now_as_utc(),
+        access_ends=now_as_utc(),
+        full_user_name="Dr. Jane Doe",
+        request_created=now_as_utc(),
+        status=AccessRequestStatus.PENDING,
+        status_changed=None,
+        changed_by=None,
     )
+
+    expected_payload = AccessRequestDetails(
+        user_id=request.user_id, dataset_id=request.dataset_id
+    ).model_dump()
+
+    expected_type = ""
+    if status == "created":
+        expected_type = dummy_config.access_request_created_type
+        await event_publisher.publish_request_created(request=request)
+    elif status == "allowed":
+        expected_type = dummy_config.access_request_allowed_type
+        await event_publisher.publish_request_allowed(request=request)
+    elif status == "denied":
+        expected_type = dummy_config.access_request_denied_type
+        await event_publisher.publish_request_denied(request=request)
+    else:
+        raise RuntimeError("Double-check test parameters")
```

Comment:
```suggestion
    publish_method = getattr(event_publisher, f"publish_request_{status}")
    await publish_method(request=request)

    expected_topic = dummy_config.access_request_events_topic
    expected_type = getattr(dummy_config, f"access_request_{status}_type")
    expected_payload = AccessRequestDetails(
        user_id=request.user_id, dataset_id=request.dataset_id
    ).model_dump()
```

this is a bit simpler, and you also get an error if there is a wrong test parameter